rocket-chatter-ingestion-server/src/process/prepare/processor/core/typeAnnotation.ts

```diff
@@ -0,0 +1,72 @@
+import { namedTypes } from "ast-types"
+
+export namespace TypeAnnotation {
+	export function flatten(
+		type: namedTypes.TSTypeAliasDeclaration["typeAnnotation"]
+	) {
+		const typeArguments = new Set<string>()
+
+		// Handle type annotations
+		if (namedTypes.TSTypeAnnotation.check(type)) {
+			flatten((type as any).typeAnnotation).forEach((x) => typeArguments.add(x))
+		}
+		// Handle direct type references
+		else if (namedTypes.TSTypeReference.check(type)) {
```

Comment:
let's modify multiple else if blocks into switch case for better readability

Reply:
Actually, it's not possible here as in each if-else ladder different expression is used rather than different comparisons on a single expression.